### PR TITLE
elasticsearch7: wrap elasticsearch-keystore

### PR DESCRIPTION
**Pull request by happysalada**
###### Motivation for this change

elasticsearch has a keystore binary that needs to be wrapped.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by happysalada** [comment]
I've just finished testing this and now with those two fixes, elastic is not complaining anymore.

I'm not sure about using sd inside the elasticsearch preStart, however substituteInPlace wasn't available.

The other thing that I am not sure is about using /var/lib/elasticsearch for logs.

I'm leaving this open for a bit in case anyone is interested.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 136255` at 6947f13f run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>elasticsearch7</li>
    <li>elasticsearch7-oss</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Consider adding a comment explaining the purpose of this patch on the line preceeding.
Near pkgs/servers/search/elasticsearch/7.x.nix:40:15:
```
   |
40 |   patches = [ ./es-home-6.x.patch ];
   |               ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/servers/search/elasticsearch/7.x.nix:56:3:
```
   |
56 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: nixos/modules/services/search/elasticsearch.nix [hunk 1]
```diff
@@ -146,8 +147,8 @@ in {
       path = [ pkgs.inetutils ];
       environment = {
         ES_HOME = cfg.dataDir;
-        ES_JAVA_OPTS = toString ( optional (!es6) [ "-Des.path.conf=${configDir}" ]
-                                  ++ cfg.extraJavaOptions);
+        ES_JAVA_OPTS = toString (optional (!es6) [ "-Des.path.conf=${configDir}" ]
+          ++ cfg.extraJavaOptions);
```
Comment (doronbehar) [inline nixos/modules/services/search/elasticsearch.nix]: Are these formatting changes necessary? It'd be easier to review this with all of them removed / part of a single commit with the title: `elasticsearch7: formatting changes`.

File: nixos/modules/services/search/elasticsearch.nix [hunk 2]
```diff
@@ -116,20 +117,20 @@ in {
...
-      default = [];
+      default = [ ];
       type = types.listOf types.str;
       example = [ "-Djava.net.preferIPv4Stack=true" ];
     };
 
     plugins = mkOption {
       description = "Extra elasticsearch plugins";
-      default = [];
+      default = [ ];
```
Comment (doronbehar) [inline nixos/modules/services/search/elasticsearch.nix:131]: Formatting changes like this as well.

File: nixos/modules/services/search/elasticsearch.nix [hunk 3]
```diff
@@ -147,9 +145,7 @@ in
       path = [ pkgs.inetutils ];
       environment = {
         ES_HOME = cfg.dataDir;
-        ES_JAVA_OPTS = toString (optional (!es6) [ "-Des.path.conf=${configDir}" ]
-          ++ cfg.extraJavaOptions);
-      } // optionalAttrs es6 {
+        ES_JAVA_OPTS = toString cfg.extraJavaOptions;
```
Comment (doronbehar) [inline nixos/modules/services/search/elasticsearch.nix:148]: The formatting done in the previous commit is especially not needed if this part gets removed eventually.

File: nixos/modules/services/search/elasticsearch.nix (hunk 1 above)
Comment (happysalada) [reply nixos/modules/services/search/elasticsearch.nix]: My bad, the formatter ran automatically and I didn't catch it.
Let me make it into a separate commit

Comment (happysalada) [reply nixos/modules/services/search/elasticsearch.nix]: I separated into 2 commits. Let me know if anything else.

Comment (doronbehar) [reply nixos/modules/services/search/elasticsearch.nix]: I don't know enough about the package to approve, sorry.

Comment (happysalada) [reply nixos/modules/services/search/elasticsearch.nix]: No problems! Thank you for having taken a look!

**Comment by happysalada** [comment]
Since this PR actually fixes the running of elastic search, I'm going to go ahead and merge this.
I'm happy to take criticism later on and make follow up PR should any other problems arise.

**Pull request merged**